src/create-context.js

```diff
@@ -40,17 +40,22 @@ export function createContext(defaultValue, contextId) {
 						// 	c.context[contextId] = _props.value;
 						// 	enqueueRender(c);
 						// });
-						subs.some(enqueueRender);
+						subs.some(c => {
+							if (!c[1] || c[1](_props.value) !== c[1](this.props.value))
+								enqueueRender(c[0])
+						});
 					}
 				};
 
-				this.sub = c => {
-					subs.push(c);
+				this.sub = (c, selector) => {
+					const entry = [c, selector]
+					const newLength = subs.push(entry);
 					let old = c.componentWillUnmount;
 					c.componentWillUnmount = () => {
-						subs.splice(subs.indexOf(c), 1);
+						subs.splice(newLength - 1, 1);
```

Comment:
Not sure if it is save to cache the position in the subscription array. If another subscription is removed the array will change and `newLength` may shift by one, effectively pointing to the wrong index.